### ISSUE TITLE
crash in metrics Cocoa method

### ISSUE THREAD

**Issue by incanus** [mapbox/mapbox-gl-native#1271]
Only seeing this on an older iPad 3, but I'm consistently getting a crash inside of our metrics check into Core Telephony. 

![screen shot 2015-04-14 at 4 35 08 pm](https://cloud.githubusercontent.com/assets/17722/7149602/612f1702-e2c4-11e4-81dc-65e90ff1f549.png)

https://github.com/mapbox/mapbox-gl-native/blob/88b8d452b9ed2cc6c156b6b446904b86afd3c7a2/platform/ios/MGLMapboxEvents.m#L574

If I had to guess, the API isn't thread safe (or shouldn't be treated as such) and we need to break out this check to the main thread similar to how we check for push. 

/cc @bleege

**Comment by incanus** [comment]
Weird, for whatever reason it only happens when debug mode is on (the menu toggle which shows tile boundaries, among other things).

**Comment by 1ec5** [comment]
`-currentRadioAccessTechnology` appears to be undocumented (but publicly exposed). There's also `-cachedCurrentRadioAccessTechnology`, which may be of interest.

**Comment by 1ec5** [comment]
Oh, that isn't the crashing thread. The crashing thread usually has a warning symbol at the top of the stack trace in the Issue navigator.

**Comment by 1ec5** [comment]
Red #1251

**Comment by bleege** [comment]
Either way I'm bummed to still see the `new` statement in there.  I'd converted it to `alloc init` to follow @1ec5's advice but it looks like it never found it's way into the code base.  :-(

Time to fix this.

**Comment by bleege** [comment]
`alloc init` has been merged.

**Comment by 1ec5** [comment]
@incanus Can you reproduce this crash still? If so, please paste in the crashing thread’s stack. Otherwise, I don’t think we can take any further action on this ticket.

**Comment by friedbunny** [comment]
iPad Mini (wifi) doesn't crash and `telephonyInfo.currentRadioAccessTechnology` returns `nil` for everything.

**Comment by bleege** [comment]
Yeah, Apple seems to be in the middle of rearranging their CoreTelephony framework.  That said these methods and constants are [still in the latest API and have been updated for Swift support](https://developer.apple.com/library/prerelease/ios/releasenotes/General/iOS83APIDiffs/modules/CoreTelephony.html).

```
@property (nonatomic, readonly, retain) NSString* currentRadioAccessTechnology __OSX_AVAILABLE_STARTING(__MAC_NA,__IPHONE_7_0);
```

I just did some testing in Debug mode on both an iPod Touch and it correctly returned `nil` for this and on my iPhone 5s where it returned `CTRadioAccessTechnologyHSDPA`.

Based on this and @1ec5's reference to #1251I think we're good to close this issue.  Before we do though let's see if @incanus can still reproduce this on the iPad 3 with the latest code fixes.
